### PR TITLE
Document x86_64-fortanix-unknown-sgx platform

### PR DESCRIPTION
**Pull request by jethrogb**
cc @raoulstrackx @mzohreva

### CODE REVIEW

File: src/doc/rustc/src/platform-support/x86_64-fortanix-unknown-sgx.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+## Testing
+
+The Rust test suite as well as custom unit and integration tests will run on
+hardware that has Intel SGX enabled if a cargo runner is configured correctly,
+see the requirements section.
+
+## Cross-compilation toolchains and C code
+
+C code is nog generally supported, as there is no libc. C code compiled for x86-64 in freestanding mode using the System V AMD64 ABI may work. The [rs-libc](https://crates.io/crates/rs-libc) crate contains a subset of libc that's known to work with this target.
```
Comment (raoulstrackx) [inline src/doc/rustc/src/platform-support/x86_64-fortanix-unknown-sgx.md:58]: Typo: `nog`

File: src/doc/rustc/src/platform-support/x86_64-fortanix-unknown-sgx.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+[ftxsgx-runner](https://crates.io/crates/fortanix-sgx-tools).
+
+See the [EDP installation
+guide](https://edp.fortanix.com/docs/installation/guide/) for recommendations
+on how to setup a development and runtime environment.
+
+## Building the target
+
+As a tier 2 target, the target is built by the Rust project.
+
```
Comment (JohnTitor) [inline src/doc/rustc/src/platform-support/x86_64-fortanix-unknown-sgx.md:45]: Could you provide a TOML example like other targets?

File: src/doc/rustc/src/platform-support/x86_64-fortanix-unknown-sgx.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+## Testing
+
+The Rust test suite as well as custom unit and integration tests will run on
+hardware that has Intel SGX enabled if a cargo runner is configured correctly,
+see the requirements section.
+
+## Cross-compilation toolchains and C code
+
+C code is not generally supported, as there is no libc. C code compiled for x86-64 in freestanding mode using the System V AMD64 ABI may work. The [rs-libc](https://crates.io/crates/rs-libc) crate contains a subset of libc that's known to work with this target.
```
Comment (JohnTitor) [inline src/doc/rustc/src/platform-support/x86_64-fortanix-unknown-sgx.md:58]: This line is a bit long, could you break it like below?
```suggestion
C code is not generally supported, as there is no libc.
C code compiled for x86-64 in freestanding mode using the System V AMD64 ABI may work.
The [rs-libc](https://crates.io/crates/rs-libc) crate contains a subset of libc that's known to work with this target.
```

**Comment by jethrogb** [comment]
Updated

**Review by JohnTitor — APPROVED** [review]

**Comment by JohnTitor** [comment]
Oh sorry, forgot to r+, thanks!

@bors r+ rollup

**Pull request merged**